### ISSUE TITLE
boto_s3_bucket.present not idempotent on ACL

### ISSUE THREAD

**Issue by kakwa** [saltstack/salt#36439]
### Description:
The boto_s3_bucket.present state is not idempotent with default values.

### Setup

(note that ACL is not set)

```yaml
my_bucket:
  boto_s3_bucket.present:
    - name: some_bucket
    - Bucket: some_bucket
```

At each run the resource is marked as modified:

```yaml
----------
          ID: my_bucket
    Function: boto_s3_bucket.present
        Name: some_bucket
      Result: True
     Comment: S3 bucket some_bucket is present.
     Started: 12:45:47.671116
    Duration: 1303.78 ms
     Changes:   
              ----------
              new:
                  ----------
                  ACL:
                      ----------
                      ACL:
                          private
              old:
                  ----------
                  ACL:
                      ----------
                      Grants:
                          |_
                            ----------
                            Grantee:
                                ----------
                                DisplayName:
                                    account
                                ID:
                                    3132323936323834000a
                                Type:
                                    CanonicalUser
                            Permission:
                                FULL_CONTROL
                      Owner:
                          ----------
                          DisplayName:
                              account
                          ID:
                              3132323936323834000a
```
### Steps to Reproduce Issue
Just apply the previously declared step.

### Versions Report
Salt Version:
           Salt: 2016.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: 0.6.4
      gitpython: 2.0.2
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.3.7
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Mar  1 2015, 12:57:24)
   python-gnupg: 0.3.6
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2
            ZMQ: 4.0.5
 
System Versions:
           dist: debian 8.4 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.4

**Comment by Ch3LL** [comment]
@kakwa this is a duplciate of #33754 which has been fixed in #33776 which looks like its in 2016.3.1. Can you give that a try and report back whether it worked and we can get this closed up. Thanks

**Comment by kakwa** [comment]
Humm, sorry, I forgot to report back.

It indeed works with the latest 2016.3 release.
